api/api-artifacts-v1/src/test/groovy/com/thoughtworks/go/apiv1/artifacts/ArtifactsControllerV1Test.groovy

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.artifacts
+
+import com.thoughtworks.go.api.SecurityTestTrait
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper
+import com.thoughtworks.go.config.*
+import com.thoughtworks.go.server.service.EntityHashingService
+import com.thoughtworks.go.server.service.ServerConfigService
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult
+import com.thoughtworks.go.server.service.result.LocalizedOperationResult
+import com.thoughtworks.go.spark.AdminUserSecurity
+import com.thoughtworks.go.spark.ControllerTrait
+import com.thoughtworks.go.spark.SecurityServiceTrait
+import groovy.json.JsonBuilder
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.invocation.InvocationOnMock
+
+import static com.thoughtworks.go.api.base.JsonUtils.toObjectString
+import static com.thoughtworks.go.apiv1.artifacts.represernter.ArtifactConfigRepresenter.toJSON
+import static org.mockito.Mockito.*
+import static org.mockito.MockitoAnnotations.initMocks
+
+class ArtifactsControllerV1Test implements SecurityServiceTrait, ControllerTrait<ArtifactsControllerV1> {
```

Comment:
Since `purgeSettings` is not mandatory, can a test be added omitting the same for show and update?